acceptance/launcher_test.go

```diff
@@ -0,0 +1,215 @@
+			}
+			expected := "bp executable"
+			if !strings.Contains(string(output), expected) {
+				t.Fatalf("failed to execute provided CMD:\n\t got: %s\n\t want: %s", output, expected)
+			}
+		})
+	})
+
+	when("CMD provided with --", func() {
+		it("launches command directly", func() {
+			cmd := exec.Command("docker", "run", "--rm", launchImage, "--", "echo", "my-cmd")
+			output, err := cmd.CombinedOutput()
+			if err != nil {
+				t.Fatalf("failed to run %v\n OUTPUT: %s\n ERROR: %s\n", cmd.Args, output, err)
+			}
+			expected := "my-cmd"
+			if !strings.Contains(string(output), expected) {
+				t.Fatalf("failed to execute provided CMD:\n\t got: %s\n\t want: %s", output, expected)
+			}
+		})
+
+		it("sets env vars from layers", func() {
+			cmd := exec.Command("docker", "run", "--rm", launchImage, "--", "env")
+			output, err := cmd.CombinedOutput()
+			if err != nil {
+				t.Fatalf("failed to run %v\n OUTPUT: %s\n ERROR: %s\n", cmd.Args, output, err)
+			}
+			expected := "SOME_VAR=some-bp-val\nOTHER_VAR=other-bp-val"
+			if !strings.Contains(string(output), expected) {
+				t.Fatalf("failed to execute provided CMD:\n\t got: %s\n\t want: %s", output, expected)
+			}
+		})
+
+		it("passes through env vars from user, excluding blacklist", func() {
+			cmd := exec.Command("docker", "run", "--rm", "-eCNB_APP_DIR=/workspace", "-eSOME_USER_VAR=some-user-val", launchImage, "--", "env")
+			output, err := cmd.CombinedOutput()
+			if err != nil {
+				t.Fatalf("failed to run %v\n OUTPUT: %s\n ERROR: %s\n", cmd.Args, output, err)
+			}
+			expected := "SOME_USER_VAR=some-user-val"
+			if !strings.Contains(string(output), expected) {
+				t.Fatalf("failed to execute provided CMD:\n\t got: %s\n\t want: %s", output, expected)
+			}
+
+			if strings.Contains(string(output), "CNB_APP_DIR") {
+				t.Fatalf("env contained white listed env far CNB_APP_DIR:\n\t got: %s\n", output)
+			}
+		})
+
+		it("adds buildpack bin dirs to the path before looking up command", func() {
+			cmd := exec.Command("docker", "run", "--rm", launchImage, "--", "bp-executable")
+			output, err := cmd.CombinedOutput()
+			if err != nil {
+				t.Fatalf("failed to run %v\n OUTPUT: %s\n ERROR: %s\n", cmd.Args, output, err)
+			}
+			expected := "bp executable"
+			if !strings.Contains(string(output), expected) {
+				t.Fatalf("failed to execute provided CMD:\n\t got: %s\n\t want: %s", output, expected)
+			}
+		})
+	})
+}
+
+func buildLaunchImage(t *testing.T) {
+	cmd := exec.Command("docker", "build", "-t", launchImage, launchDockerContext)
+	if output, err := cmd.CombinedOutput(); err != nil {
+		t.Fatalf("failed to run %v\n OUTPUT: %s\n ERROR: %s\n", cmd.Args, string(output), err)
+	}
+}
+
+func removeLaunchImage(t *testing.T) {
+	cmd := exec.Command("docker", "rmi", launchImage)
+	if output, err := cmd.CombinedOutput(); err != nil {
+		t.Fatalf("failed to run %v\n OUTPUT: %s\n ERROR: %s\n", cmd.Args, string(output), err)
+	}
+}
+
+func buildLauncher(t *testing.T) {
```

Comment:
Is there a reason why we're not making use of `buildBinaries` from acceptance_test.go?

Reply:
We only need the `launcher` for this test, and building just the launcher is considerably faster (it doesn't need to be compiled in a container).